documentation/git.md

```diff
@@ -0,0 +1,142 @@
+
+More explanation, for example, [here at SourceTree](https://www.atlassian.com/blog/sourcetree/interactive-rebase-sourcetree).
+
+## Keep up with the main branch
+
+Before making a pull request, ensure your branch is as close to the main one as possible to avoid unnecessary merge conflicts.
+
+First, ensure that your local main branch is current, e.g. through `git fetch origin main:main` ([source](https://stackoverflow.com/a/17722977/2169092)).
+
+### Rebase
+
+This can be done with a rebase: `git rebase main`.
+
+It applies every commit from your branch on top of the main branch. If successful, the entire branch will appear to have started on the most recent state of the main branch.
+
+There may be merge conflicts. As rebasing handles commits individually, one merge conflict may need to be resolved multiple times. After resolving a conflict, do `git rebase --continue`. If you get stuck, roll back everything with `git rebase --abort`.
+
+If this branch already exists on the remote, you must force-push the rebase, overwriting the remote branch with your local one. This can be done safely via `git push origin --force-with-lease`. The option prevents you from unintentionally overwriting someone else’s work. If others have pushed to the branch, contact them.
+
+### Merge
+
+An alternative is to merge the main branch into yours: `git pull origin main`. This will also ensure the local main branch is current.
+
+That adds one commit to your branch, containing all changes since the first divergence of your branches with the main one. If merge conflicts arise, they can be more complex than with a rebase, but each can be resolved at once.
+
+### How to choose
+
+Rebasing can be more laborious and has a risk of being destructive. It does keep the history in Git easier to comprehend because of its more linear form.
+
+Merging often makes the tree resemble a shunting yard with connections upstream and downstream. It takes more time to locate branches and commits within them. Merge commits repeat changes already present in others.
+
+Neither approach is mandatory or prohibited. It is good to continue evaluating with colleagues who work within the same repository.
+
+## Git CLI shortcuts
+
+For users of Git via the CLI, it may be useful to create aliases for some commonly used git commands. You can save this in a configuration file of your shell. For example:
+
+```bash
+alias gcm="git checkout main"
+alias gcp="git checkout -"
+alias gh="git push"
+alias ghf="git push origin --force-with-lease"
+alias gl="git log -3"
+alias gp="git fetch --prune; git pull"
+alias gpb="git branch --merged | grep -v "\*" | grep -Ev "(\*|master|develop)" | xargs -n 1 git branch -d"
+alias gpr="git pull --rebase"
+alias gra="git rebase --abort"
+alias grc="git rebase --continue"
+alias grm="git rebase main"
+alias gs="git status"
```

Comment:
`gs` is a cli command for ghost script, and I think there a couple more commands in your aliases. So I don't think it's good practice to suggest this big list of aliases. They might cause weird conflicts and could be hard to memorize.